### ISSUE TITLE
Pipeline Debug Rule Metrics are not removed from prometheus exporter when enabled and then disabled

### ISSUE THREAD

**Issue by drewmiranda-gl** [Graylog2/graylog2-server#16311]
Pipeline Debug Rule Metrics are not removed from prometheus exporter when enabled and then disabled

Although the pipeline debug metrics stop updating, the last value is persisted in the prometheus exporter and causes a flat line

![image](https://github.com/Graylog2/graylog2-server/assets/107503402/eb0d0bfa-9f5d-444a-a682-1c532ce25e96)

## Expected Behavior

Disabled metrics are removed from prometheus exporter

## Current Behavior

Metrics enabled and then disabled leave record in prometheus that never update. The only way to clear them is to restart graylog-server (or stop/start)

## Steps to Reproduce (for bugs)

Prerequisites: graylog server, graylog prom exporter enabled/configured, prometheus, grafana

1. Enable pipeline debug metrics (System / Pipelines / Manage rules)
2.Disable pipeline debug metrics
3. Observe that metric in prometheus exporter never changes and is not removed

## Your Environment

* Graylog Version: Graylog 5.1.4+6fa2de3
* Java Version: Bundled
* OpenSearch Version: 2.8.0
* MongoDB Version: 6.0.9
* Operating System: Ubuntu Server 22.04 LTS
* Browser version: Google Chrome Version 116.0.5845.110 (Official Build) (arm64)

Please let me know if there are any questions

**Comment by drewmiranda-gl** [comment]
stale, won't fix
